Lib/test/test_zoneinfo/test_zoneinfo.py

```diff
@@ -1507,6 +1530,26 @@ def test_clear_cache_two_keys(self):
         self.assertIsNot(dub0, dub1)
         self.assertIs(tok0, tok1)
 
+    def test_clear_cache_refleak(self):
+        class Stringy(str):
+            allow_comparisons = True
+            def __eq__(self, other):
+                if not self.allow_comparisons:
+                    raise CustomError
+                return super().__eq__(other)
+            __hash__ = str.__hash__
+
+        key = Stringy("America/Los_Angeles")
+        self.klass(key)
```

Comment:
To increase the chance that this actually works:

```suggestion
        self.klass.clear_cache(only_keys="America/Los_Angeles")
        self.klass(key)
```